src/dbnode/server/server.go

```diff
@@ -600,9 +600,9 @@ func Run(runOpts RunOptions) {
 		// SetDatabase() once we've initialized it.
 		service = ttnode.NewService(nil, ttopts)
 	)
-	if cfg.Tchannel.MaxIdleTime > 0 && cfg.Tchannel.IdleCheckInterval > 0 {
-		tchannelOpts.MaxIdleTime = cfg.Tchannel.MaxIdleTime
-		tchannelOpts.IdleCheckInterval = cfg.Tchannel.IdleCheckInterval
+	if cfg.TChannel != nil {
+		tchannelOpts.MaxIdleTime = cfg.TChannel.MaxIdleTime
+		tchannelOpts.IdleCheckInterval = cfg.TChannel.IdleCheckInterval
```

Comment:
Do these need to be greater than 0?

Reply:
There are checks in the tchannel code already.
```
func (o *ChannelOptions) validateIdleCheck() error {
        if o.IdleCheckInterval > 0 && o.MaxIdleTime <= 0 {
                return errMaxIdleTimeNotSet
        }

        return nil
}
```
```
func (is *idleSweep) start() {
        if is.started || is.idleCheckInterval <= 0 {
                return
        }

        is.ch.log.WithFields(
                LogField{"idleCheckInterval", is.idleCheckInterval},
                LogField{"maxIdleTime", is.maxIdleTime},
        ).Info("Starting idle connections poller.")

        is.started = true
        is.stopCh = make(chan struct{})
        go is.pollerLoop()
}
```